src/MetadataManager/EventManager.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+
+import {Balloon} from '../Utils/Balloon';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+import {Logger} from '../Utils/Logger';
+
+import {Metadata} from './metadataAPI';
+import {PathToHash} from './pathToHash';
+
+
+class EventQueue {
+  private inProgress: boolean = false;
+  private queue: {method: any, input: {[key: string]: any}}[] = [];
+
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(method: any, input: {[key: string]: any}): void {
+    this.queue.push({method: method, input: input});
+    this.autoAction();
+  }
+
+  front() {
+    return this.queue[0];
+  }
+  dequeue() {
+    this.queue.shift();
+  }
+
+  clear() {
+    this.inProgress = false;
+    this.queue = [];
+  }
+
+  isEmpty() {
+    return this.queue.length === 0;
+  }
+
+  autoAction() {
+    if (this.inProgress === false) {
+      this.inProgress = true;
+      this.action();
+    }
+  }
+
+  async action() {
+    await this.front().method();
+    this.dequeue();
+
+    if (this.isEmpty()) {
+      this.clear();
+    } else {
+      this.action();
+    }
+  }
+}
+
+class EventBuffer {
+  private _queue = new EventQueue();
+  constructor() {}
+  public setEvent(request: any, input: {[key: string]: any}) {
+    this._queue.enqueue(() => {
+      return new Promise(() => {
+        if (Object.keys(input).length === 0) {
+          request();
+        } else {
+          request();
+        }
+      });
+    }, input);
+  }
+}
+
+
+
+/* istanbul ignore next */
+export class MetadataEventManager {
+  private fileWatcher = vscode.workspace.createFileSystemWatcher(`**/*`);  // glob pattern
+  private eventBuffer = new EventBuffer();
+
+  public static didCreateUri: vscode.Uri|undefined = undefined;
+
+  public static register(context: vscode.ExtensionContext) {
+    let workspaceRoot: vscode.Uri|undefined = undefined;
+
```

Comment:
Maybe here..?

https://github.com/Samsung/ONE-vscode/pull/1288/files#r984225593

If PathToHash is used only here.